src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -40,9 +40,34 @@
 
 /**
  * <p>
- * Checks that a class that has only private constructors and has no descendant
- * classes is declared as final.
+ * Check that identifies classes that can be declared as final are explicitly made final.
+ * The following is the type of classes that can be identified:
  * </p>
+ * <ol>
+ *   <li>
+ *       Private classes with implicit constructor.
+ *   </li>
+ *   <li>
+ *       Classes with any modifier, and contains only private constructors.
+ *   </li>
+ * </ol>
+ * <p>
+ * Classes are skipped if:
+ * </p>
+ * <ol>
+ *   <li>
+ *       Class is already {@code final}.
+ *   </li>
+ *   <li>
+ *       Class is declared as {@code abstract}.
+ *   </li>
+ *   <li>
+ *       Class is Super class of some Anonymous inner class.
+ *   </li>
```

Comment:
Really, it seems like this whole section is unnecessary since it can be summed up as "we don't place violations on code that already meets criteria (class is final) or is not compilable", since both conditions are already implied by (a) all checks work like this, and (b) Checkstyle should not be ran on code that doesn't compile (this is already documented elsewhere).

Reply:
@nrmancuso  should we remove the entire section ?

Reply:
Probably, @rnveach any thoughts on this?

Reply:
I was the one who requested this list. 

Some things weren't clear and once I asked more about one of the items, I think its like a bug. https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208582723
I don't see `Class is Super class of some Anonymous inner class` being clear to me.

The first 2 just make sense since we started a list, but I am ok with them removed.

Reply:
> I don't see Class is Super class of some Anonymous inner class being clear to me.

```
kevin@inspiron-15-5510:~/Desktop/check_style$ cat Test.java
public class Test {
    private final class SuperClass { 
        private SuperClass() {
        }    
    }

    SuperClass SC = new SuperClass() {
    };
}
kevin@inspiron-15-5510:~/Desktop/check_style$ javac Test.java
Test.java:7: error: cannot inherit from final Test.SuperClass
    SuperClass SC = new SuperClass() {
                        ^
1 error
kevin@inspiron-15-5510:~/Desktop/check_style$ 
```

@rnveach I hope this makes you clear

Reply:
Sorry, I meant clear in terms that I would have known this without documentation saying it. We won't print violations on already final/abstract classes was clear even without us specifically saying it.

Reply:
@rnveach should we remove this list as this only mention think mentioned at https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208560428

Reply:
@rnveach ping

Reply:
I am unsure what response is requested from me besides https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208598751 . Again, I requested the table because the list wasn't fully clear to me. I am fine with removing the obvious things like `final` and `abstract`. But the others and this issue were not as obvious.

If needs be, feel free to IM me.

Reply:
@Kevin222004 please remove obvious items as described at https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1217231774 and address other items so we can get this merged.

Reply:
@Kevin222004 ping

Reply:
I have removed the final and abstract one.

I have chnaged the sentence as `Class is extended by another class in the same file`

https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1217231774

https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208598751 
@rnveach again the purpose of 
> I don't see Class is Super class of some Anonymous inner class being clear to me. https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208598751 

is to show https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1208603089

Reply:
One mistake that I have made is, have blindly copy paste some names without thinking in docs which is misleading in code as well https://github.com/checkstyle/checkstyle/pull/12737#discussion_r1228969922

Reply:
@nrmancuso @rnveach ping lets complete this

Reply:
@rnveach ping please continue